aesara/link/numba/dispatch.py

```diff
@@ -0,0 +1,263 @@
+
+    subtensor_def_src = f"""
+def subtensor({", ".join(input_names)}):
+    {indices_creation_src}
+    {index_body}
+    return z
+    """
+
+    return subtensor_def_src
+
+
+@numba_funcify.register(Subtensor)
+@numba_funcify.register(AdvancedSubtensor)
+@numba_funcify.register(AdvancedSubtensor1)
+def numba_funcify_Subtensor(op, node, **kwargs):
+
+    idx_list = getattr(op, "idx_list", None)
+    subtensor_def_src = create_index_func(
+        node, idx_list, objmode=isinstance(op, AdvancedSubtensor)
+    )
+
+    global_env = {}
+    global_env["objmode"] = numba.objmode
+
+    subtensor_fn = compile_function_src(subtensor_def_src, "subtensor", global_env)
+
+    return numba.njit(subtensor_fn)
+
+
+@numba_funcify.register(DeepCopyOp)
+def numba_funcify_DeepCopyOp(op, node, **kwargs):
+
+    # Scalars are apparently returned as actual Python scalar types and not
+    # NumPy scalars, so we need two separate Numba functions for each case.
+    if node.outputs[0].type.ndim == 0:
+        # TODO: Do we really need to compile a pass-through function like this?
+        @numba.njit
+        def deepcopyop(x):
+            return x
+
+    else:
+
+        @numba.njit
+        def deepcopyop(x):
+            return x.copy()
+
+    return deepcopyop
+
+
+@numba_funcify.register(MakeSlice)
+def numba_funcify_MakeSlice(op, **kwargs):
+    # XXX: This won't work when calling into object mode (e.g. for advanced
+    # indexing), because there's no Numba unboxing for its native `slice`
+    # objects.
+
+    @numba.njit
+    def makeslice(*x):
+        return slice(*x)
+
+    return makeslice
+
+
+def _register_scalar_ops():
+    _scalar_ops = {
+        aes.Abs: np.abs,
+        aes.Angle: np.angle,
+        aes.ArcCos: np.arccos,
+        aes.ArcCosh: np.arccosh,
+        aes.ArcSin: np.arcsin,
+        aes.ArcSinh: np.arcsinh,
+        aes.ArcTan: np.arctan,
+        aes.ArcTanh: np.arctan2,
+        aes.Ceil: np.ceil,
+        aes.Conj: np.conj,
+        aes.Cos: np.cos,
+        aes.Cosh: np.cosh,
+        aes.Deg2Rad: np.deg2rad,
+        aes.Erf: scipy.special.erf,
+        aes.Erfc: scipy.special.erfc,
+        aes.Erfcinv: scipy.special.erfcinv,
+        aes.Erfcx: scipy.special.erfcx,
+        aes.Erfinv: scipy.special.erfinv,
+        aes.Exp: np.exp,
+        aes.Exp2: np.exp2,
+        aes.Expm1: np.expm1,
+        aes.Floor: np.floor,
+        aes.Gamma: scipy.special.gamma,
+        aes.GammaLn: scipy.special.gammaln,
+        aes.I0: scipy.special.i0,
+        aes.I1: scipy.special.i1,
+        aes.Imag: np.imag,
+        aes.IsInf: np.isinf,
+        aes.IsNan: np.isnan,
+        aes.J0: scipy.special.j0,
+        aes.J1: scipy.special.j1,
+        aes.Log: np.log,
+        aes.Log10: np.log10,
+        aes.Log1p: np.log1p,
+        aes.Log2: np.log2,
+        aes.Psi: scipy.special.psi,
+        aes.Rad2Deg: np.rad2deg,
+        aes.Sgn: np.sign,
+        aes.Sin: np.sin,
+        aes.Sinh: np.sinh,
```

Comment:
I don't think we need these mappings, because they should be picked up automatically from the `nfunc_specs`&mdash;like they are in the JAX `dispatch` module.

Also, we might need to use "primitive" version of many of these basic math functions in order to make good use of `numba.vectorize`.